src/rt/sched/threadpool.h

```diff
@@ -61,17 +64,15 @@ namespace verona::rt
     ThreadSync<T> sync;
 #endif
 
-    std::atomic_uint64_t barrier_count = 0;
     uint64_t barrier_incarnation = 0;
 
-    T* first_thread = nullptr;
+    /// List of instantiated scheduler threads.
+    /// Contains both free and active threads; protects accesses with a lock.
+    SchedulerList<T>* threads = nullptr;
```

Comment:
Could this be a `SchedulerList<T>`, then we don't have to worry about its lifetime? Rather than a separate allocation.

Reply:
I get an error from snmalloc:
```
error: variable does not have a constant initializer
 SNMALLOC_REQUIRE_CONSTINIT static ThreadPool<T, C> global_thread_pool;
```

Reply:
Done.